### PR TITLE
refactor: rename XR to XRSystem

### LINKED ISSUE #25732: Rename XR to XRSystem
Our `XR` interface should be named `XRSystem`, a change made in https://github.com/immersive-web/webxr/pull/948

This involves:

 - Renaming `dom/webidls/XR.webidl`
 - Renaming `dom/xr.rs`
 - Renaming `XR`, `XRBinding`, `XRMethods` to `XRSystem*` instead

https://github.com/servo/servo/blob/ba23c5afee380930bd0f2c5fd665d318443ed19f/components/script/dom/xr.rs#L41

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR renames XR to XRSystem.

---

- [x] `./mach build -d` does not report any errors 
- [x] `./mach test-tidy` does not report any errors
- [X] These changes fix #25732 

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because it's the changing of a name (XR -> XRSystem)

### CODE REVIEW

**Comment by jsjoeio** [comment]
Running into issues during the build process :( 

When I run:

```sh
pip install virtualenv
```

I get this:

```
Requirement already satisfied: virtualenv in /usr/local/lib/python3.7/site-packages (20.0.2)
Requirement already satisfied: six<2,>=1.12.0 in /usr/local/lib/python3.7/site-packages (from virtualenv) (1.14.0)
Requirement already satisfied: distlib<1,>=0.3.0 in /usr/local/lib/python3.7/site-packages (from virtualenv) (0.3.
0)
Requirement already satisfied: filelock<4,>=3.0.0 in /usr/local/lib/python3.7/site-packages (from virtualenv) (3.0
.12)
Requirement already satisfied: importlib-metadata<2,>=0.12; python_version < "3.8" in /usr/local/lib/python3.7/sit
e-packages (from virtualenv) (1.5.0)
Requirement already satisfied: appdirs<2,>=1.4.3 in /usr/local/lib/python3.7/site-packages (from virtualenv) (1.4.
3)
Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.7/site-packages (from importlib-metadata<2,>=0.
12; python_version < "3.8"->virtualenv) (2.2.0)
```

Then I run this `./mach build -d` and get this:
```
Python virtualenv is not installed. Please install it prior to running mach.
```

I haven't dabbled in Python in a while so forgive me if this is a n00b question, but what am I doing wrong? It feels like a never ending loop 😂

P.S. - if you do help, or if I figure it out, we can update the README with a "Troubleshooting a Mac environment" section ❤️

**Comment by Manishearth** [comment]
Can you remove the brewfile changes? Those are for CI. We can land this and perhaps open a second PR to improve docs on this.

**Comment by jdm** [comment]
With respect to the pip issue, try `pip2 install virtualenv` or some other binary that explicitly references the python2.7 pip binary, since pip appears to be part of the python3 distribution.

**Comment by jsjoeio** [comment]
Okay, I removed the brewfile changes. 

CI is still failing - but is that simply because I didn't write any unit tests? Looking at the logs, it seems like it's printing the diffs but I don't see any error messages or explanation as to why the task is failing.

**Comment by paulrouget** [comment]
It’s a formatting issue.

Run `mach fmt`.

File: components/script/dom/xr.rs [hunk 1]
```diff
@@ -4,9 +4,9 @@
```
Comment (jsjoeio) [inline components/script/dom/xr.rs]: Should this file be called `xrsystem.rs` now?

File: components/script/dom/webidls/XR.webidl [hunk 2]
```diff
@@ -2,9 +2,9 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
```
Comment (jsjoeio) [inline components/script/dom/webidls/XR.webidl]: Should this file be called `XRSystem.webidl`?

**Comment by jsjoeio** [comment]
Here's the error from the task:
```shell
error[E0583]: file not found for module `XRSystemBinding`
   --> /repo/target/debug/build/script-bbd50c7c9ef66328/out/Bindings/mod.rs:382:9
    |
382 | pub mod XRSystemBinding;
    |         ^^^^^^^^^^^^^^^
    |
    = help: name the file either XRSystemBinding.rs or XRSystemBinding/mod.rs
```

So it doesn't like the name...best way to fix this?

File: components/script/dom/webidls/XR.webidl (hunk 2 above)
Comment (jdm) [reply components/script/dom/webidls/XR.webidl]: Yes, that will avoid the compile problem.

File: components/script/dom/xr.rs (hunk 1 above)
Comment (jdm) [reply components/script/dom/xr.rs]: Yes, and you will need to change the reference in components/script/dom/mod.rs as well.

**Comment by jsjoeio** [comment]
Question: is it possible to set up a pre-push hook that runs `./mach fmt`? Wouldn't that save time?

**Comment by jsjoeio** [comment]
Okay...now this error is a bit more cryptic to me:

```shell
error[E0050]: method `RequestSession` has 4 parameters but the declaration in trait `dom::bindings::codegen::Bindings::XRSystemBinding::XRSystemBinding::XRSystemMethods::RequestSession` has 3
   --> components/script/dom/xrsystem.rs:165:9
    |
165 | /         &self,
166 | |         mode: XRSessionMode,
167 | |         init: RootedTraceableBox<XRSessionInit>,
168 | |         comp: InRealm,
    | |_____________________^ expected 3 parameters, found 4
    | 
   ::: /repo/target/debug/build/script-bbd50c7c9ef66328/out/Bindings/XRSystemBinding.rs:893:23
    |
893 |       fn RequestSession(&self, mode: XRSessionMode, parameters: RootedTraceableBox<crate::dom::bindings::codegen::Bindings::XRSystemBinding::XRSessionInit>) -> Rc<Promise>;
    |                         ----------------------------------------------------------------------------------------------------------------------------------- trait requires 3 parameters
error: aborting due to previous error
```

Any suggestions on how to resolve this?

**Comment by jdm** [comment]
Ah, that's probably https://github.com/servo/servo/blob/master/components/script/dom/bindings/codegen/Bindings.conf#L143.

**Review by Manishearth — APPROVED** [review]
Please squash your commits

**Comment by jsjoeio** [comment]
Thanks for approving! I'll squash now 👍🏼

I'm curious though, what's the advantage to having authors squash their commits as opposed to [squashing as part of the PR merge process](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#squash-and-merge-your-pull-request-commits)?

**Comment by jdm** [comment]
This doesn't build yet:
```
error[E0405]: cannot find trait `XRMethods` in this scope
   --> components/script/dom/xrsystem.rs:119:6
    |
119 | impl XRMethods for XRSystem {
    |      ^^^^^^^^^ not found in this scope
```
It needs to implement (and import) XRSystemMethods instead.

As for the question about squashing, we use a bot that performs merges for us and it doesn't support squashing yet.

**Comment by jsjoeio** [comment]
image

Wow. That was more than I expected 😅 First intro to OSS with Rust. Thanks for having beginner issues and providing guidance/help.

Glad this is finally passing. Should be ready to go 👍🏼

**Comment by Manishearth** [comment]
@bors-servo r+

Thanks!

**Comment by Manishearth** [comment]
Looks like idlharness hasn't been updated upstream.


<details>

```
  │ 
  │ IdlInterface.prototype.assert_interface_object_exists@https://web-platform.test:8443/resources/idlharness.js:1388:5
  │ IdlInterface.prototype.test_self/<@https://web-platform.test:8443/resources/idlharness.js:1599:14
  │ Test.prototype.step@https://web-platform.test:8443/resources/testharness.js:2024:25
  │ test@https://web-platform.test:8443/resources/testharness.js:548:30
  │ self.subsetTestByKey@https://web-platform.test:8443/resources/idlharness.js:54:14
  │ IdlInterface.prototype.test_self@https://web-platform.test:8443/resources/idlharness.js:1579:5
  │ IdlInterface.prototype.test@https://web-platform.test:8443/resources/idlharness.js:1554:14
  │ IdlArray.prototype.test@https://web-platform.test:8443/resources/idlharness.js:888:28
  └ idl_test/</<@https://web-platform.test:8443/resources/idlharness.js:3391:31
  ▶ Unexpected subtest result in /webxr/idlharness.https.window.html:
  │ FAIL [expected PASS] Stringification of navigator.xr
  │   → assert_class_string: class string of navigator.xr expected "[object XR]" but got "[object XRSystem]"
  │ 
  │ IdlInterface.prototype.test_primary_interface_of/<@https://web-platform.test:8443/resources/idlharness.js:2795:9
  │ Test.prototype.step@https://web-platform.test:8443/resources/testharness.js:2024:25
  │ test@https://web-platform.test:8443/resources/testharness.js:548:30
  │ self.subsetTestByKey@https://web-platform.test:8443/resources/idlharness.js:54:14
  │ IdlInterface.prototype.test_primary_interface_of@https://web-platform.test:8443/resources/idlharness.js:2791:5
  │ IdlInterface.prototype.test_object@https://web-platform.test:8443/resources/idlharness.js:2724:10
  │ IdlArray.prototype.test/<@https://web-platform.test:8443/resources/idlharness.js:900:36
  │ IdlArray.prototype.test@https://web-platform.test:8443/resources/idlharness.js:895:21
  └ idl_test/</<@https://web-platform.test:8443/resources/idlharness.js:3391:31
  ▶ Unexpected subtest result in /webxr/idlharness.https.window.html:
  │ FAIL [expected PASS] XR interface: existence and properties of interface prototype object
  │   → assert_own_property: self does not have own property "XR" expected property "XR" missing
  │ 
  │ IdlInterface.prototype.assert_interface_object_exists@https://web-platform.test:8443/resources/idlharness.js:1388:5
  │ IdlInterface.prototype.test_self/<@https://web-platform.test:8443/resources/idlharness.js:1906:14
  │ Test.prototype.step@https://web-platform.test:8443/resources/testharness.js:2024:25
  │ test@https://web-platform.test:8443/resources/testharness.js:548:30
  │ self.subsetTestByKey@https://web-platform.test:8443/resources/idlharness.js:54:14
  │ IdlInterface.prototype.test_self@https://web-platform.test:8443/resources/idlharness.js:1897:5
  │ IdlInterface.prototype.test@https://web-platform.test:8443/resources/idlharness.js:1554:14
  │ IdlArray.prototype.test@https://web-platform.test:8443/resources/idlharness.js:888:28
  └ idl_test/</<@https://web-platform.test:8443/resources/idlharness.js:3391:31

```

</details>

**Comment by Manishearth** [comment]
For now i'd rerun the test and update expectations, and allow it to be fixed upstream.

**Comment by jdm** [comment]
No need to change the test expectations now that #25783 merged.
@bors-servo retry

**Comment by jdm** [comment]
Ok, I was wrong. There are two cases of test expectations and tests that need to be adjusted:
```
  ▶ Unexpected subtest result in /webxr/idlharness.https.window.html:
  └ PASS [expected FAIL] XRSystem interface: existence and properties of interface object
  ▶ Unexpected subtest result in /webxr/idlharness.https.window.html:
  └ PASS [expected FAIL] XRSystem interface object length
  ▶ Unexpected subtest result in /webxr/idlharness.https.window.html:
  └ PASS [expected FAIL] XRSystem interface object name
  ▶ Unexpected subtest result in /webxr/idlharness.https.window.html:
  └ PASS [expected FAIL] XRSystem interface: existence and properties of interface prototype object
  ▶ Unexpected subtest result in /webxr/idlharness.https.window.html:
  └ PASS [expected FAIL] XRSystem interface: existence and properties of interface prototype object's "constructor" property
  ▶ Unexpected subtest result in /webxr/idlharness.https.window.html:
  └ PASS [expected FAIL] XRSystem interface: existence and properties of interface prototype object's @@unscopables property
```
Means that those corresponding entries in [this file](https://github.com/servo/servo/blob/master/tests/wpt/metadata/webxr/idlharness.https.window.js.ini#L134-L156) need to be removed.

```
  ▶ Unexpected subtest result in /_mozilla/mozilla/interfaces.html:
  │ FAIL [expected PASS] Interfaces exposed on the window
  │   → assert_true: If this is failing: DANGER, are you sure you want to expose the new interface XRSystem to all webpages as a property on the global? Do not make a change to this file without review from jdm or Ms2ger for that specific change! expected true got false
  │ 
  │ test_interfaces/<@http://web-platform.test:8000/_mozilla/mozilla/interfaces.js:72:7
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1867:25
  │ test@http://web-platform.test:8000/resources/testharness.js:548:30
  │ test_interfaces@http://web-platform.test:8000/_mozilla/mozilla/interfaces.js:2:3
  └ @http://web-platform.test:8000/_mozilla/mozilla/interfaces.html:13:1
```
This means that [this line](https://github.com/servo/servo/blob/master/tests/wpt/mozilla/tests/mozilla/interfaces.html#L266) needs to be updated for the new interface name (and please reorder alphabetically afterwards).

**Comment by jsjoeio** [comment]
Thanks for the specific guidance on how to update the tests! I did as followed but something is failing now. I can't figure it out. The [logs](https://community-tc.services.mozilla.com/tasks/KI1DFvs6TESzgjIocPSxFw/runs/0/logs/https%3A%2F%2Fcommunity-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FKI1DFvs6TESzgjIocPSxFw%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log) are vague:
> [taskcluster 2020-02-22 18:38:23.121Z] === Task Finished ===
[taskcluster 2020-02-22 18:38:23.122Z] Unsuccessful task run with exit code: 1 completed in 773.899 seconds

**Comment by Manishearth** [comment]
```
 0:08.38(B INFO(B Diffing old and new manifests /repo/tests/wpt/mozilla/meta/MANIFEST.json
 0:08.67(B WARNING(B Manifest /repo/tests/wpt/mozilla/meta/MANIFEST.json contains correct tests but file hashes changed.
 0:08.67(B ERROR(B Manifest /repo/tests/wpt/mozilla/meta/MANIFEST.json is outdated, use |./mach update-manifest| to fix.
 0:08.67(B INFO(B Diffing old and new manifests /repo/tests/wpt/webgl/meta/MANIFEST.json
 0:09.76(B INFO(B Diffing old and new manifests /repo/tests/wpt/metadata/MANIFEST.json

```

**Comment by jsjoeio** [comment]
meme: not sure if nothing is broken or if everything is broken

**Comment by CYBAI** [comment]
@bors-servo retry

-  #24622,  #25476

**Pull request merged**